Docs: Field config docs update, Table docs update, Override matcher naming issue

torkelo opened the pull request:
* Updates field config doc page with minor edits making it more accurate and clear (I hope). 
* Moves table options to the table page 
* Filters => Matchers rename 

Discovered a big discrepancy in the field override matcher naming. 
In the dropdown we refer to them as "Filter ...." 
<img width="356" alt="Screen Shot 2020-09-12 at 16 48 32" src="https://user-images.githubusercontent.com/10999/92998045-d4b66e80-f517-11ea-963d-998682f4f92c.png">

After adding them we mix the word with matcher & filter:
<img width="364" alt="Screen Shot 2020-09-12 at 16 48 57" src="https://user-images.githubusercontent.com/10999/92998061-e861d500-f517-11ea-82a5-e5e2b2c7e356.png">

New proposal:
![Screenshot from 2020-09-14 20-35-50](https://user-images.githubusercontent.com/10999/93124511-fc4d3880-f6c9-11ea-9060-a1d822bc8ec8.png)


The docs are updated to use the word matcher instead. 

Fixes #27388

torkelo committed:
Docs: Field Config Docs Update, Table Docs update, Override matcher naming sync

torkelo committed:
removed sentance that feels duplicated

dprokop reviewed (approved):
I think this looks good. I can't recall why exactly we have used _filter_ rather than _matcher_ in the copy, I think @oddlittlebird will have some input on this.

oddlittlebird commented:
Because at one point we were going to change "matcher" to "filter" in the UI because "matcher" isn't really a word unless you are using it in the context of marriage. It is a synonym for "matchmaker."

oddlittlebird commented on docs/sources/panels/field-configuration-options.md:
```diff
@@ -83,19 +79,19 @@ Field overrides allow you to change the settings for one field (column in tables
 1. When finished, click **Save** to save all panel edits to the dashboard.
 
-## Filter options
+## Override matchers
```
Same comment as above. "matcher" is not a correct word. People understand what "filters" are. We will need to explain "matcher" and it will still look wrong.

oddlittlebird commented on docs/sources/panels/field-configuration-options.md:
```diff
@@ -105,37 +101,6 @@ This section explains all available field options. They are listed in alphabetic
 
 Most field options will not affect the visualization until you click outside of the field option box you are editing or press Enter.
 
-### Cell display mode
```
Why was this content removed?

torkelo replied:
it was moved to the table visualization, as these options are all table specific display options (custom field config options)

oddlittlebird commented on docs/sources/panels/field-configuration-options.md:
```diff
@@ -261,7 +226,7 @@ Let’s assume that our result set is a data frame that consists of four fields:
 | 2020-01-02 03:06:00 | 48.0      | 31.0     | 68       |
 
-Let's apply the field options from the [field option example](#field-option-example) to apply the Celsius unit and get rid of the decimal place. This results in the following table:
+Let's apply the field options from the [field option example](#ield-option-example) to apply the Celsius unit and get rid of the decimal place. This results in the following table:
```
```suggestion
Let's apply the field options from the [field option example](#field-option-example) to apply the Celsius unit and get rid of the decimal place. This results in the following table:
```

oddlittlebird commented on docs/sources/panels/visualizations/table-panel.md:
```diff
@@ -27,6 +27,43 @@ Table visualizations allow you to apply:
 
 - **Show header -** Show or hide column names imported from your data source..
 
+### Custom field options
```
This content does not belong here. Please remove.

torkelo replied:
It does. These are table visualization settings. I think they should be together with the other table display settings.

ryantxu replied:
I just proposed https://github.com/grafana/grafana/pull/27579 to help make this more clear in the UI :)

oddlittlebird commented on docs/sources/panels/field-configuration-options.md:
```diff
@@ -83,19 +79,19 @@ Field overrides allow you to change the settings for one field (column in tables
+This section explains all available matchers for field overrides. They are listed in alphabetical order.
 
-### Filter field by name
+### Match field by name
```
Please change these headings back. The original headings align more closely to the UI text.

torkelo replied:
I did update the UI as well, as using the Filter word was a bit confusing. But think I have a better naming than match.

Add override dropdown:
* Fields matching name
* Fields matching regex
* Fields with type

oddlittlebird replied:
Works for me.

oddlittlebird commented on docs/sources/panels/field-configuration-options.md:
```diff
@@ -83,19 +79,19 @@ Field overrides allow you to change the settings for one field (column in tables
+## Override matchers
 
-This section explains all available filter options for field overrides. They are listed in alphabetical order.
+This section explains all available matchers for field overrides. They are listed in alphabetical order.
```
Again, "matcher" is not the correct word in this context. They are options or filter options.

ryantxu replied:
:+1: we need to find a good explanation/heading for this.  The basic concept is that that there are setting that apply to fields -- the "override" section lets you selectively apply new settings to the results.

torkelo commented on docs/sources/panels/field-configuration-options.md:
```diff
@@ -8,23 +8,24 @@ weight = 300
+This page explains what field options and field overrides in Grafana are and how to use them. It also includes
+[examples](#examples) if you need an idea of how this feature might be useful in the real world.
 
-> **Note:** This documentation refers to a Grafana 7.0 beta feature. This documentation will be frequently updated to reflect updates to the feature, and it will probably be broken into smaller sections when the feature moves out of beta.
```
No it's not

oddlittlebird replied:
I wish someone had mentioned that to me earlier. I will add breaking up this topic into coherent chunks to my To Do list.

torkelo commented on docs/sources/panels/field-configuration-options.md:
```diff
@@ -8,23 +8,24 @@ weight = 300
 
-Field configuration options allow you to change how the data is displayed in your visualizations. Options and overrides that you apply do not change the data, they change how Grafana displays the data.
-
-> **Note:** The time fields are not affected by field configuration options or overrides.
```
Yes they are

torkelo commented on docs/sources/panels/field-configuration-options.md:
```diff
@@ -8,23 +8,24 @@ weight = 300
 
-_Field configuration options_, both standard and custom, are applied in the **Field** tab in the panel editor. Changes on this panel apply to all fields in the visualization. For example, if you change the unit to percentage, then all fields with numeric values will be displayed in percentages. [Apply a field option](#configure-all-fields).
+_Field configuration options_, both standard and custom, can be found in the **Field** tab in the panel editor. Changes
+on this tab apply to all fields (i.e. series/columns). For example, if you change the unit to percentage,
```
Sorry, to my editor does this automatically sometimes to make all text visible. The newline s do not show in the tendered page only in markdown

oddlittlebird replied:
Any chance you could disable that? I know it looks fine in the output, but it makes mid-sentence line breaks and, as mentioned, is a PAIN TO EDIT. It makes more work for the writers.

torkelo replied:
sorry, restored the newlines

torkelo committed:
Update docs/sources/panels/field-configuration-options.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

torkelo committed:
Update docs/sources/panels/field-configuration-options.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

torkelo committed:
Update docs/sources/panels/field-configuration-options.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

ryantxu commented on docs/sources/panels/field-configuration-options.md:
```diff
@@ -42,26 +43,19 @@ Standard field options are:
-- [Gauge]({{< relref "visualizations/gauge-panel.md" >}})
-- [Stat]({{< relref "visualizations/stat-panel.md" >}})
-- [Table]({{< relref "visualizations/table-panel.md" >}})
+You can apply standard field options to most of the built-in Grafana panels. Some older panels and community panels that have
```
The standard options apply to every visualization -- for the built in visualizations everything shown is used.  For older/community panels it really depends on how they implemented it.

torkelo replied:
So instead of listing all core visualizations (except heatmap) I figured it's better to explain why core panels use has field options and some don't.

ryantxu commented on docs/sources/panels/field-configuration-options.md:
```diff
@@ -42,26 +43,19 @@ Standard field options are:
-
-Custom field options are:
-
-- [Column width](#column-width)
```
custom options can be applied to *any* visualization.  Table was the only built-in visualization using custom options, but any visualization can add config that should be applied to a set of fields

torkelo committed:
Update docs/sources/panels/field-configuration-options.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

ryantxu commented:
What about "Override rule"?  does that better capture what we are doing?

torkelo committed:
Merge remote-tracking branch 'origin/master' into field-config-docs-update

torkelo committed:
removed bad newlines and minor tweaks, still need to figure out naming for matchers in UI and docs

torkelo committed:
Updated matcher names and descriptions, and UX

torkelo commented:
Updated the UI, back to Override N for titles as using the "matcher" name there was a big confusing. 

Also updated the matchers names and descriptions. 

![Screenshot from 2020-09-14 20-35-50](https://user-images.githubusercontent.com/10999/93124679-433b2e00-f6ca-11ea-9f73-8e3451a429d7.png)

Need to sync the docs with this now

torkelo committed:
Updated field override docs

torkelo committed:
Fixed plural

torkelo committed:
Updated wording

torkelo committed:
removed plurals for the selectors heading

torkelo committed:
Updated names

ryantxu reviewed (approved):
Removing "matcher" from the UI helps!  

I think there is more we can do to help explain what is actually happening with fields/overrides -- but this is a big improvement.

torkelo committed:
Updated docs
